### PR TITLE
Design rules refactor

### PR DESCRIPTION
**Pull request by tschesky**
Refactored the whole RoomsController to fit with the Ruby Community Design Rules.
This mostly includes making smaller methods, moving some re-usable blocks of code to new methods etc.
Needs testing on the live application.
The only thing left to do is the ABC metric on report_issue method.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -9,7 +9,15 @@ default: &default
   timeout: 5000
 
 development:
-  <<: *default
+  adapter: postgresql
```
Comment (NynaPykulska) [inline config/database.yml:12]: Don't put your local settings on master.

File: app/controllers/rooms_controller.rb [hunk 2]
```diff
@@ -1,115 +1,167 @@
...
+  def fetch_issues_by_room_id
+    @issues_for_room = Issue.where('room_id = ?', params[:room_id])
+  end
+
+  def fetch_all_rooms
+    @rooms = Room.all
+  end
+
+  def fetch_selected_issues
+    @selected_issues = Issue.where(room_id: params[:id])
```
Comment (NynaPykulska) [inline app/controllers/rooms_controller.rb:106]: Can we use one convention?
Either

>  Issue.where('room_id = ?', params[:room_id]) 

or

>  Issue.where(room_id: params[:id])

Also, use either room_id or id

File: app/controllers/rooms_controller.rb [hunk 3]
```diff
@@ -1,115 +1,167 @@
...
+      unless Issue.exists?(room_id: id, issue_type_id: issue.to_i)
+        Issue.create(room_id: id.to_i, issue_type_id: issue.to_i,
+                     priority: 'Medium', is_done: true,
+                     is_recurring: false)
+      end
+    end
+  end
+
+  def remove_old_issues_from_room(selected_issues, id)
+    @issues_for_room = Issue.where('room_id = ?', id)
```
Comment (NynaPykulska) [inline app/controllers/rooms_controller.rb:127]: Use already existing method fetch_issues_by_room_id which does exactly that.

**Comment by NynaPykulska** [comment]
Oi. Comments.

File: config/database.yml (hunk 1 above)
Comment (tschesky) [reply config/database.yml:12]: K, I won't - then GL testing on the master DB.

File: app/controllers/rooms_controller.rb (hunk 2 above)
Comment (tschesky) [reply app/controllers/rooms_controller.rb:106]: Yes, we can - I'll put it in another refactor. It is, however, out of scope for this one as I also need to adjust the parameters passed from views.

File: app/controllers/rooms_controller.rb (hunk 3 above)
Comment (tschesky) [reply app/controllers/rooms_controller.rb:127]: The fetch_issues_by_room_id is a before_action fetch - it's OK to use it in a view where I need a pre-loaded array/value etc. but I don't see the point in calling it in a re-usable method. We have the ID passed by argument as there is no guarantee that in every place I call it - the room ID will be conveniently stored in the params[] table and will always by called :room_id.

**Pull request merged**